nxdrive/client/local/linux.py

```diff
@@ -67,7 +68,22 @@ def remove_remote_id_impl(path: Path, name: str = "ndrive") -> None:
     def set_folder_icon(self, ref: Path, icon: Path) -> None:
         """Use commandline to customize the folder icon."""
         log.debug(f"Setting the folder icon of {ref!r} using {icon!r}")
+
+        shared_icons_dir = Path("~/.local/share/icons/")
         folder_path = self.abspath(ref).as_posix()
+
+        # Create the shared icons folder if it doesn't already exist
+        shared_icons_dir.mkdir(parents=True, exist_ok=True)
+
+        # Emblems icons must be saved in ~/.local/share/icons/ to be accessible
+        try:
+            emblem_path = shared_icons_dir / "emblem-nuxeo.svg"
+            if emblem_path.is_file():
```

Comment:
This should not be necessary as the check should be done in `has_folder_icon()`.
Here you should just to the copy without checks.